src/domain/UBGraphicsScene.cpp

```diff
@@ -3122,3 +3146,42 @@ void UBGraphicsScene::initStroke()
 {
     mCurrentStroke = new UBGraphicsStroke(shared_from_this());
 }
+
+void UBGraphicsScene::MultiTouchDrawing(QTouchEvent* event, UBStylusTool::Enum currentTool)
+{
+    QList <QTouchEvent::TouchPoint> touchPoints = event->touchPoints();
+    QPointF lastPoint_m, endPoint_m;
+    foreach (QTouchEvent::TouchPoint point, touchPoints)
```

Comment:
Qt6 recommends to prefer the C++ range based `for` over the `foreach` keyword. See https://doc.qt.io/qt-6/foreach-keyword.html. It is also recommended to use a const reference in a case like this to avoid copying the point.